### PR TITLE
Autoformatting

### PR DESCRIPTION
**Pull request by rogerwaleffe**
This PR adds auto-formatters and linters for the Python and CPP sources. They have been added to GitHub actions to enforce the same style for the code base. The pip install dependencies have also been limited for simplicity.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,226 @@
+---
```
Comment (shivaram) [inline .clang-format:1]: I understand this was generated from the Google C++ style guide. Did we have to change anything or did we just use the auto-generated one directly?

File: MANIFEST.in [hunk 2]
```diff
@@ -1,10 +1,3 @@
-graft test
```
Comment (shivaram) [inline MANIFEST.in:1]: any reason these are now removed?

File: docs/examples/config/lp_paleobiology.rst [hunk 3]
```diff
@@ -159,11 +159,11 @@ The output should appear similar to::
 After this has finished, our output will be in our ``[dataset_dir]`` (using the provided config, this will be ``dataset/``.
 
 Here are the files that were created in training:
-Let's check again what was added in the ``datasets/custom_lp_example/`` directory. For clarity, we only list the files that were created in training. Notice that several files have been created, including the trained model, the embedding table, a full configuration file, and output logs:
```
Comment (shivaram) [inline docs/examples/config/lp_paleobiology.rst:162]: Is removing custom_lp_example supposed to be a part of this PR?

**Review by shivaram — COMMENTED** [review]
Thanks @rogerwaleffe - I had some minor comments. Also could you retrigger the tests that failed previously?

File: .clang-format (hunk 1 above)
Comment (JasonMoho) [reply .clang-format:1]: This used the auto-generated one directly.

File: MANIFEST.in (hunk 2 above)
Comment (JasonMoho) [reply MANIFEST.in:1]: Yes, these files were not needed for building and installing the package. They were left in from the original packaging implementation. This just cleans that up.

File: docs/examples/config/lp_paleobiology.rst (hunk 3 above)
Comment (JasonMoho) [reply docs/examples/config/lp_paleobiology.rst:162]: So this PR includes the documentation changes from #89. Bad practice to include that here. Can keep or remove.

**Review by JasonMoho — APPROVED** [review]

**Comment by cthoyt** [comment]
amazinnggggg

**Pull request merged**